code/modules/research/techweb/__techweb_helpers.dm

```diff
@@ -168,16 +174,33 @@
 			if(!ispath(path))
 				continue
 			if(length(SSresearch.techweb_boost_items[path]))
-				SSresearch.techweb_boost_items[path] += list(node.id = node.boost_item_paths[path])
+				SSresearch.techweb_boost_items[path][node.id] = node.boost_item_paths[path]
 			else
 				SSresearch.techweb_boost_items[path] = list(node.id = node.boost_item_paths[path])
 		CHECK_TICK
 
 /proc/techweb_item_boost_check(obj/item/I)			//Returns an associative list of techweb node datums with values of the boost it gives.	var/list/returned = list()
 	if(SSresearch.techweb_boost_items[I.type])
-		return SSresearch.techweb_boost_items[I.type]		//It should already be formatted in node datum = value.
+		return SSresearch.techweb_boost_items[I.type]		//It should already be formatted in node datum = list(point type = value)
 
 /proc/techweb_item_point_check(obj/item/I)
 	if(SSresearch.techweb_point_items[I.type])
 		return SSresearch.techweb_point_items[I.type]
-	return 0
+
+/proc/techweb_point_display_generic(pointlist)
+	var/list/ret = list()
+	for(var/i in pointlist)
+		if(SSresearch.point_types[i])
+			ret += "[SSresearch.point_types[i]]: [pointlist[i]]"
+		else
+			ret += "ERRORED POINT TYPE: [pointlist[i]]"
+	return ret.Join("<BR>")
+
+/proc/techweb_point_display_rdconsole(pointlist, last_pointlist)
+	var/list/ret = list()
+	for(var/i in pointlist)
+		if(SSresearch.point_types[i])
+			ret += "[SSresearch.point_types[i]]: [pointlist[i]] (+[last_pointlist[i]])"
+		else
+			ret += "ERRORED POINT TYPE: [pointlist[i]] (+[last_pointlist[i]])"

```

Comment:
Can simplify this and above with something like `"[SSresearch.point_types[i] || "Error"]: ..."`